Modules/ITS/src/ITSFeeTask.cxx

```diff
@@ -0,0 +1,201 @@
+    for (int i = 0; i < mNTrigger; i++) {
+      mErrorFlag->GetXaxis()->SetBinLabel(i + 1, mErrorType[i]);
+    }
+  }
+  if (mTFInfo) {
+    setAxisTitle(mTFInfo, "TF ID", "Counts");
+  }
+  if (mErrorFlagVsFeeId) {
+    setAxisTitle(mErrorFlagVsFeeId, "FeeID", "Error Flag ID");
+    for (int i = 0; i < mNTrigger; i++) {
+      mErrorFlagVsFeeId->GetYaxis()->SetBinLabel(i + 1, mErrorType[i]);
+    }
+  }
+}
+
+void ITSFeeTask::startOfActivity(Activity& /*activity*/)
+{
+  QcInfoLogger::GetInstance() << "startOfActivity" << AliceO2::InfoLogger::InfoLogger::endm;
+}
+
+void ITSFeeTask::startOfCycle() { QcInfoLogger::GetInstance() << "startOfCycle" << AliceO2::InfoLogger::InfoLogger::endm; }
+
+void ITSFeeTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  std::chrono::time_point<std::chrono::high_resolution_clock> start;
+  std::chrono::time_point<std::chrono::high_resolution_clock> end;
+  int difference;
+  start = std::chrono::high_resolution_clock::now();
+
+  DPLRawParser parser(ctx.inputs());
+  int staveStart = 0;
+  for (int i = 0; !mEnableLayers[i]; i++) {
+    staveStart = StaveBoundary[i + 1];
+  }
+  for (auto it = parser.begin(), end = parser.end(); it != end; ++it) {
+    auto const* rdh = it.get_if<o2::header::RAWDataHeaderV6>();
+    int istave = (int)(rdh->feeId & 0x00ff);
+    int ilink = (int)((rdh->feeId & 0x0f00) >> 8);
+    istave += staveStart;
+
+    if ((int)(rdh->stop) && it.size()) { //looking into the Stop RDH
+      ILOG(Info) << "Stopbit : " << (int)rdh->stop << ", Payload size: " << it.size() << ENDM;
+      auto const* ddw = reinterpret_cast<const GBTDdw*>(it.data());
+      LOGF(INFO, "DDW header 0x: %02x", ddw->ddwBits.data8[9]);
+    }
+    for (int i = 0; i < 13; i++) { //filling trigger types to debug the program for the moment
+      if (((uint32_t)(rdh->triggerType) >> i & 1) == 1) {
+        mErrorFlag->Fill(i + 1);
+        mErrorFlagVsFeeId->Fill((istave * 3) + ilink, i + 1);
+      }
+    }
+  }
+
+  mTimeFrameId = ctx.inputs().get<int>("G");
+
+  mTFInfo->Fill(mTimeFrameId);
+  end = std::chrono::high_resolution_clock::now();
+  difference = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
+  ILOG(Info) << "Processing time: " << difference << ", and TF ID == " << mTimeFrameId << ENDM;
+}
+
+void ITSFeeTask::getEnableLayers()
+{
+  std::ifstream configFile("Config/ConfigFakeRateFee.dat");
+  for (int ilayer = 0; ilayer < NLayer; ilayer++) {
+    configFile >> mEnableLayers[ilayer];
+    if (mEnableLayers[ilayer]) {
+      ILOG(Info) << "Enable layer : " << ilayer << ENDM;
+    }
+  }
+  configFile >> mRunNumberPath;
+}
+
+void ITSFeeTask::endOfCycle()
+{
+  std::ifstream runNumberFile("/home/its/QC_Online/workdir/infiles/RunNumber.dat"); //catching ITS run number in commissioning
+  if (runNumberFile) {
+    std::string runNumber;
+    runNumberFile >> runNumber;
+    ILOG(Info) << "runNumber : " << runNumber << ENDM;
+    if (runNumber == mRunNumber) {
```

Comment:
perhaps `if (runNumber != mRunNumber){
    getObjectsManager()->addMetadata(mTFInfo->GetName(), "Run", runNumber);
    getObjectsManager()->addMetadata(mErrorFlagVsFeeId->GetName(), "Run", runNumber);
    getObjectsManager()->addMetadata(mErrorFlag->GetName(), "Run", runNumber);
    mRunNumber = runNumber;
}
I think that it is logically the same but it avoids the `goto`.

Reply:
removed. no need to make the checking here.